modules/nodes-nodes-configuring-graceful-shutdown.adoc

```diff
@@ -0,0 +1,119 @@
+
+[NOTE]
+====
+If you do not configure graceful node shutdown, the default grace period is `0` and the pod is forcefully evicted from the node.
+====
+
+.Prerequisites
+
+* You have access to the cluster with the `cluster-admin` role.
+* You have defined priority classes for pods that require critical or regular classification.
+
+.Procedure
+
+. Define shutdown grace periods in the `KubeletConfig` CR by saving the following YAML in the `kubelet-gns.yaml` file:
++
+[source,yaml]
+----
+apiVersion: machineconfiguration.openshift.io/v1
+kind: KubeletConfig
+metadata:
+  name: graceful-shutdown
+  namespace: openshift-machine-config-operator
+spec:
+  machineConfigPoolSelector:
+    matchLabels:
+      pools.operator.machineconfiguration.openshift.io/worker: "" <1>
+  kubeletConfig:
+    shutdownGracePeriod: "3m" <2>
+    shutdownGracePeriodCriticalPods: "2m" <3>
+----
+<1> This example applies shutdown grace periods to nodes with the `worker` role. 
+<2> Define a time period for regular pods to shut down.
+<3> Define a time period for critical pods to shut down. 
+
+. Create the KubeletConfig CR by running the following command:
++
+[source,terminal]
+----
+$ oc create -f kubelet-gns.yaml
+----
++
+.Example output
+[source,terminal]
+----
+kubeletconfig.machineconfiguration.openshift.io/graceful-shutdown created
+----
+
+.Verification
+
+. View the kubelet logs for a node to verify the grace period configuration by using the command line or by viewing the `kublet.conf` file.
+
+.. To view the logs by using the command line, run the following command, replacing `<node_name>` with the name of the node:
++
+[source,bash]
+----
+$ oc adm node-logs <node_name> -u kubelet
+----
++
+.Example output
+[source,terminal]
+----
+Sep 12 22:13:46
+ci-ln-qv5pvzk-72292-xvkd9-worker-a-dmbr4
+hyperkube[22317]: I0912 22:13:46.687472
+22317 nodeshutdown_manager_linux.go:134]
+"Creating node shutdown manager"
+shutdownGracePeriodRequested="3m0s"
+shutdownGracePeriodCriticalPods="2m0s"
+shutdownGracePeriodByPodPriority=[
+{Priority:0
+ShutdownGracePeriodSeconds:1200}
+{Priority:2000000000
+ShutdownGracePeriodSeconds:600}]
+...
+----
++
+Ensure that the log messages for `shutdownGracePeriodRequested` and `shutdownGracePeriodCriticalPods` match the values set in the `KubeletConfig` CR. 
++
+.. To view the logs in the `kublet.conf` file on a node, run the following commands to enter a debug session on the node and view the `kublet.conf` file:
++
+[source,terminal]
+----
+$ oc debug node/<node_name>
+----
++
+[source,terminal]
+----
+$ chroot /host
+----
++
+[source,terminal]
+----
+$ cat /etc/kubernetes/kubelet.conf 
+----
++
+Ensure that the log messages for `shutdownGracePeriodRequested` and `shutdownGracePeriodCriticalPods` match the values set in the `KubeletConfig` CR.
+
+. During a graceful node shutdown, you can verify that a pod was gracefully shut down by running the following command, replacing `<pod_name>` with the name of the pod:
```

Comment:
```suggestion
. Verify that a pod was terminated by running the following command, replacing `<pod_name>` with the name of the pod:
```

Reply:
I added the "During a graceful node shutdown... " part because I think you can only see the status of pods while the node is still in the process of shutting down. After the node shuts down, you won't see the pod because it won't exist AFAIK. So perhaps I'll keep this bit in.